### ISSUE TITLE
emptiness TTL is not applied to instance type g4dn

### ISSUE THREAD

**Issue by vi-credo** [aws/karpenter#2085]
## Version

**Karpenter**: v0.13.1

**Kubernetes**: v1.22

## Expected Behavior

`ttlSecondsAfterEmpty` parameter is set to 30 seconds. Emptiness TTL is applied to the node once no workload is running on it. After 30 seconds node is removed.

Works for instance type `t2.medium`!

## Actual Behavior

For instance type `g4dn.2xlarge` node was not removed after being empty for 30 seconds. I encountered same behaviour for other sizes of the same instance family. Also `controller.node Added TTL to empty node` log was not present, assuming TTL was not assigned in the first place.

## Steps to Reproduce the Problem

Create a provisioner, then create a deployment with a matching nodeSelector, and delete in about 1-2 minutes. Please, see manifests below.

## Resource Specs and Logs

Provisioner (with AWSNodeTemplate):
```yaml
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: test
spec:
  labels:
    env: test
  ttlSecondsAfterEmpty: 30
  requirements:
    - key: node.kubernetes.io/instance-type
      operator: In
      values:
        - g4dn.2xlarge
  providerRef:
    name: test
---
apiVersion: karpenter.k8s.aws/v1alpha1
kind: AWSNodeTemplate
metadata:
  name: test
spec:
  amiFamily: AL2
  subnetSelector:
    karpenter.sh/discovery: test-cluster
  securityGroupSelector:
    karpenter.sh/discovery: test-cluster
  tags:
    karpenter.sh/discovery: test-cluster
```
Simple deployment:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: hello
  namespace: default
spec:
  replicas: 1
  selector:
    matchLabels:
      name: hello
  template:
    metadata:
      labels:
        name: hello
    spec:
      nodeSelector:
        env: test
      containers:
        - name: hello-world
          image: gcr.io/google-samples/hello-app:1.0
```

**Comment by tzneal** [comment]
Are  you running the [nvidia device plugin](https://github.com/NVIDIA/k8s-device-plugin)?  I expect we're waiting for that node to be initialized which won't occur if the nvidia device plugin daemonset doesn't run on the node to register the extended resources.

**Comment by tzneal** [comment]
Related to https://github.com/aws/karpenter/issues/2021

**Comment by vi-credo** [comment]
No, I don't have it running.

**Comment by vi-credo** [comment]
Confirmed to work with nvidia device plugin daemonset running on the GPU node. I actually wanted to avoid deploying that plugin and use Karpenter + node labels for workloads with different hardware requirements in the first place, but looks like I came back around to the initial setup with exposing GPU resources to Kubernetes. I guess this covers my case, but probably worth mentioning in Karpenter docs for GPU workloads.
